common/browser_context.go

```diff
@@ -388,23 +370,34 @@ func (b *BrowserContext) waitForEvent(event string, predicateFn goja.Callable, t
 
 	select {
 	case <-b.ctx.Done():
-		b.logger.Debugf("BrowserContext:WaitForEvent:ctx.Done", "bctxid:%v event:%q", b.id, event)
+		return nil, errors.New("test iteration ended")
```

Comment:
Could context be canceled by something else, too? I'm wondering if this error message's would be accurate then 🤔

Reply:
Good spot. I was meant to leave a WDYT note. So when the context is closed, it definitely means that iteration is ending, but it could also mean the whole test run is ending. Would you prefer the original error from the context itself?

Reply:
I agree, I don't think we should mention "iteration end". Maybe something more generic?

> Would you prefer the original error from the context itself?

What would that be?

Reply:
@ankur22, yes, the context error itself would be better 👍 So we can better diagnose when it ends because of something else.

Reply:
Resolved in dc62a318fcb0d2ccb9110fb933e30cbed02e103d